assets/js/app/Data.js

```diff
@@ -9,7 +9,13 @@ LocusZoom.DataSources = function() {
     this.sources = {};
 };
 
+LocusZoom.KnownDataSources = [];
```

Comment:
Having `KnownDataSources` and `DataSources` as separate (but tightly coupled) attributes of the core `LocusZoom` object isn't as clean as it could be. A pattern used in other places (`Transformations`, `LabelFunctions`, `DataLayers`, etc.) is to have a single object in the core that contains a structure for storing members as well as whatever methods are necessary for adding/deleting/interacting with members:

```javascript
LocusZoom.DataSources = (function() {
  var obj = {};
  var sources = {};

  obj.add = function(name, data_source) {
    ...
  };

  obj.set = function(name, data_source) {
    ...
  };

  ...

  return obj;
})();
```

A minor bonus to this is that the `DataSources` object should know what it does and does not have, so there's no need for checks against the existence of `KnownDataSources` as appears in `findKnownSource()`.